Source/Core/Common/FileUtil.cpp

```diff
@@ -276,33 +279,68 @@ bool DeleteDir(const std::string& filename)
   return false;
 }
 
+// Repeatedly invokes func until it returns true or max_attempts failures.
+// Waits after each failure, with each delay doubling in length.
+template <typename FuncType>
+static bool AttemptMaxTimesWithExponentialDelay(const FuncType& func, int max_attempts,
```

Comment:
I'd recommend having the function to retry be the last parameter to make caller sites look a bit better once formatted with clang-format.